### PR TITLE
Initialize app

### PR DESCRIPTION
**Pull request by erkanerol**
Towards https://github.com/giantswarm/roadmap/issues/752

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,67 +1,11 @@
-[![CircleCI](https://circleci.com/gh/giantswarm/{APP-NAME}-app.svg?style=shield)](https://circleci.com/gh/giantswarm/{APP-NAME}-app)
+[![CircleCI](https://circleci.com/gh/giantswarm/csi-external-snapshotter-app.svg?style=shield)](https://circleci.com/gh/giantswarm/csi-external-snapshotter-app)
 
-# {APP-NAME} chart
+# CSI External Snapshotter
 
-Giant Swarm offers a {APP-NAME} App which can be installed in workload clusters.
-Here we define the {APP-NAME} chart with its templates and default configuration.
+To be able to create volume snapshots with Cinder CSI, we need to install CRDs and snapshot controller. See
```
Comment (pipo02mix) [inline README.md:5]:
```suggestion
To be able to create volume snapshots with any CSI, we need to install CRDs and a snapshot controller. See
```
Let's make it generic

File: helm/templates/crd-install/crd-job.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      serviceAccountName: {{ include "crdInstall" . }}
+      securityContext:
+        runAsUser: 1000
+        runAsGroup: 2000
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        effect: NoSchedule
+      containers:
+      - name: kubectl
+        image: "{{ .Values.images.registry }}/giantswarm/docker-kubectl:latest"
```
Comment (pipo02mix) [inline helm/templates/crd-install/crd-job.yaml:31]: I am always in favor of specific version, but it is not a hard requirement

File: helm/templates/crd-install/crd-np.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  # allow egress traffic to the Kubernetes API
+  egress:
+  - ports:
+    - port: 443
+      protocol: TCP
+    # legacy port kept for compatibility
+    - port: 6443
+      protocol: TCP
+    to:
+    {{- range tuple "10.0.0.0/8" "172.16.0.0/12" "192.168.0.0/16" "100.64.0.0/10" }}
```
Comment (pipo02mix) [inline helm/templates/crd-install/crd-np.yaml:29]: are those always the same? should we come from values yaml?

File: helm/templates/snapshot-controller-rbac.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  name: snapshot-controller-leaderelection
+subjects:
+  - kind: ServiceAccount
+    name: snapshot-controller
+    namespace: {{ .Release.Namespace | quote }}
+roleRef:
+  kind: Role
+  name: snapshot-controller-leaderelection
+  apiGroup: rbac.authorization.k8s.io
+
```
Comment (pipo02mix) [inline helm/templates/snapshot-controller-rbac.yaml:75]:
```suggestion
```

**Review by pipo02mix — APPROVED** [review]
LGTM

File: helm/templates/crd-install/crd-np.yaml (hunk 3 above)
Comment (erkanerol) [reply helm/templates/crd-install/crd-np.yaml:29]: Honestly,  I copied/pasted crd-install manifests from other repos.

**Pull request merged**